kolibri/core/content/signals.py

```diff
@@ -22,3 +23,14 @@ def reorder_channels_upon_deletion(sender, instance=None, *args, **kwargs):
     For a given channel, decrement the order of all channels that come after this channel.
     """
     ChannelMetadata.objects.filter(order__gt=instance.order).update(order=F('order') - 1)
+
+
+@receiver(pre_delete, sender=ChannelMetadata)
+def update_lesson_resources_before_delete(sender, instance=None, *args, **kwargs):
+    # Update the resources array of all lessons to ensure they don't have
+    # any deleted content
+    lessons = Lesson.objects.all()
+    for lesson in lessons:
+        for resource in lesson.resources:
+            lesson.resources = [r for r in lesson.resources if r['channel_id'] != instance.id]
+        lesson.save()
```

Comment:
jumping into django - nice!

cc @rtibbles interested to get your eyes on this because this isn't my wheelhouse. two possible concerns:

* it re-saves every lesson whether or not the resource list has changed. A simple optimization might be to only save if the list changed
* secondly, I've read that signals in django can sometimes be [made more explicit](https://lincolnloop.com/blog/django-anti-patterns-signals/)  by putting this code in the channel metadata model